### PR TITLE
 Task 6 completed

### PR DESCRIPTION
**Pull request by Tucker-Ajai**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -83,14 +82,85 @@ describe("4. GET /api/reviews/:review_id", () => {
...
+      .expect(200)
+      .then((response) => {
+        expect(response.body.msg).toBe("No matching review id");
+      });
+  });
+});
+
+describe("6. GET /api/reviews/:review_id/comments", () => {
+  test("Function gives requested information when given the correct ID", () => {
+    return request(app)
```
Comment (Jodiech) [inline __tests__/app.test.js:142]: This test is too brittle, as you have hard-coded this. It would be better to use the test data to check that the properties match what you expect. 👍

File: __tests__/app.test.js [hunk 2]
```diff
@@ -83,14 +82,85 @@ describe("4. GET /api/reviews/:review_id", () => {
...
+        expect(response.body.reviewComments).toEqual(result);
+      });
+  });
+  test("The Recieved data is to have the most recent comments first", () => {
+    return request(app)
+      .get("/api/reviews/2/comments")
+      .expect(200)
+      .then((response) => {
+        expect(response.body.reviewComments).toBeSortedBy("created_at", {
+          descending: true,
```
Comment (Jodiech) [inline __tests__/app.test.js:163]: Nice use of jest-sorted here 😄

File: __tests__/app.test.js [hunk 3]
```diff
@@ -83,14 +82,85 @@ describe("4. GET /api/reviews/:review_id", () => {
...
+        });
+      });
+  });
+  test("When client inputs an ID that is not a number, the Function will throw an error", () => {
+    return request(app)
+      .get("/api/reviews/;Drop table/comments")
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe("Invalid review ID provided");
+      });
```
Comment (Jodiech) [inline __tests__/app.test.js:173]: I'd be a little more semantic with your test descriptions. I'd suggest writing something along the lines of Status 400: responds with an error when an invalid Id is passed.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -83,14 +82,85 @@ describe("4. GET /api/reviews/:review_id", () => {
...
+  });
+  test("When Review id is valid but there is no matching data, Function to respond appropriately ", () => {
```
Comment (Jodiech) [inline __tests__/app.test.js]: This should be more semantic, for example: Status: 200 responds with an empty array when a valid ID is passed and there a no comments.

File: __tests__/app.test.js [hunk 5]
```diff
@@ -83,14 +82,85 @@ describe("4. GET /api/reviews/:review_id", () => {
...
+    return request(app)
+      .get("/api/reviews/13/comments")
+      .expect(200)
+      .then((response) => {
+       // console.log(response.body)
```
Comment (Jodiech) [inline __tests__/app.test.js]: Please remove any console logs in your code! Keep it nice and clean 🗑️

**Review by Jodiech — COMMENTED** [review]
Hey 👋 nice work on the ticket! I have left a few comments for you, mostly regarding your testing. Please make the changes and resubmit once this is done 😄

File: __tests__/app.test.js [hunk 6]
```diff
@@ -83,14 +82,68 @@ describe("4. GET /api/reviews/:review_id", () => {
...
+  test("Function gives requested information when given the correct ID", () => {
+    return request(app)
+      .get("/api/reviews/2/comments")
+      .expect(200)
+      .then((response) => {
+        const result = testData.commentData.filter(
+          (obj) => obj.review_id === 2
+        );
+
+        const test = response.body.reviewComments.filter(
```
Comment (MrMcSnail) [inline __tests__/app.test.js:150]: Bit of a edge case use of the delete operator here and there's no need to filter. You can use assertions within an iteration. Refactor this so that you iterate through the array and assert that each element has the required properties and the correct review id.

**Review by MrMcSnail — CHANGES_REQUESTED** [review]
